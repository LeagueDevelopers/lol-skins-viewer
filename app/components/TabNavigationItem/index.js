import React, { PropTypes } from 'react';
import { pure } from 'recompose';
import { hashHistory } from 'react-router';
import cx from 'classnames';

import style from './index.scss';

function NavigationItem (props) {
  const { to, children, className, isActive } = props;
  return (
    <div className={cx(style.navigationItemWrapper, className)}>
      <div className={cx(style.navigationItem, { [style.active]: isActive })}>
        <span
          className={style.content}
          onClick={() => !isActive && hashHistory.replace(to)}
        >
          {children}
        </span>
      </div>
    </div>
  );
}

NavigationItem.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  currentPath: PropTypes.string,
  index: PropTypes.bool,
  tab: PropTypes.bool,
  isActive: PropTypes.bool
};

export default pure(NavigationItem);
