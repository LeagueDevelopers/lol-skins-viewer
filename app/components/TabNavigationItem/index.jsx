import React, { PropTypes } from 'react';
import cx from 'classnames';

import style from './index.scss';

function NavigationItem (props) {
  const { onClick, children, className, isActive } = props;
  return (
    <div className={cx(style.navigationItem, isActive && style.active, className)}>
      <div className={style.contentWrapper}>
        <span
          className={style.content}
          onClick={onClick}
        >
          {children}
        </span>
      </div>
    </div>
  );
}

NavigationItem.propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  isActive: PropTypes.bool
};

export default NavigationItem;
