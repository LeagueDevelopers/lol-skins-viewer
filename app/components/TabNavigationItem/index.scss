.navigationItem {
  display: flex;
  cursor: pointer;
  margin-right: 30px;
  min-width: 28px;
  white-space: nowrap;
  text-overflow: ellipsis;
	margin-top: 9px;

  .contentWrapper {
    position: relative;
    color: #CDBE91;
    font-size: 12px;
    font-weight: 500;
    letter-spacing: 0.075em;
    text-transform: uppercase;
    font-family: 'LoL Display';
    transition: text-shadow 0.3s ease-in-out;
    width: 100%;
    min-height: 20px;

    .content {
      display: block;
      text-align: center;
      cursor: pointer;
      line-height: normal;
    }
  }
}

.navigationItem.active {
	.contentWrapper {
    color: #F0E6D2;

    &:after {
			content: "";
			height: 1px;
			width: 100%;
			position: absolute;
			left: 0;
			top: 20px;
			margin: 0;
			background: linear-gradient(to left, transparent, #cdbe91, transparent);
		}
	}
}
