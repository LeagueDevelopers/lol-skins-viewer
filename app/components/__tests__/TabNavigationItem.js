import React from 'react';
import { shallow } from 'enzyme';

import TabNavigationItem from '../TabNavigationItem';

describe('<TabNavigationItem />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <TabNavigationItem onClick={() => false} className="testClass" isActive>
        {'Test Route'}
      </TabNavigationItem>
    );
    expect(wrapper.get(0)).toMatchSnapshot();
  });

  it('should have the .active class when isActive is true', () => {
    const wrapper = shallow(
      <TabNavigationItem onClick={() => false} isActive>
        {'Test Route'}
      </TabNavigationItem>
    );
    expect(wrapper.hasClass('active')).toBe(true);
  });

  it('should not have the .active class when isActive is falsey', () => {
    const wrapper = shallow(
      <TabNavigationItem onClick={() => false} isActive={false}>
        {'Test Route'}
      </TabNavigationItem>
    );
    expect(wrapper.hasClass('active')).toBe(false);
    wrapper.setProps({ isActive: null });
    expect(wrapper.hasClass('active')).toBe(false);
    wrapper.setProps({ isActive: undefined });
    expect(wrapper.hasClass('active')).toBe(false);
  });

  it('should render its children', () => {
    const wrapper = shallow(
      <TabNavigationItem>
        Test Route
      </TabNavigationItem>
    );
    expect(wrapper.find('.content').text()).toBe('Test Route');
  });

  it('should call onClick when its text content is clicked', () => {
    const handleClick = jest.fn();
    const wrapper = shallow(
      <TabNavigationItem onClick={handleClick} isActive={false}>
        {'Test Route'}
      </TabNavigationItem>
    );
    wrapper.find('.content').simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });
});
