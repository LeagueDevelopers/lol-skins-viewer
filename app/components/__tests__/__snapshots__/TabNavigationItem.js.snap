exports[`<TabNavigationItem /> should render correctly 1`] = `
<div
  className="navigationItem active testClass">
  <div
    className="contentWrapper">
    <span
      className="content"
      onClick={[Function]}>
      Test Route
    </span>
  </div>
</div>
`;
